rest/oidc_test_provider_test.go

```diff
@@ -39,3 +48,84 @@ func TestExtractSubjectFromRefreshToken(t *testing.T) {
 	require.NoError(t, err, "invalid refresh token error")
 	assert.Equal(t, subject, sub)
 }
+
+func TestInsecureOpenIDConnectAuth(t *testing.T) {
+	providers := auth.OIDCProviderMap{
+		"test": &auth.OIDCProvider{
+			Register:      true,
+			Issuer:        "${baseURL}/db/_oidc_testing",
+			Name:          "test",
+			ClientID:      "sync_gateway",
+			ValidationKey: base.StringPtr("qux"),
+			CallbackURL:   base.StringPtr("${baseURL}/db/_oidc_callback"),
+		},
+	}
+	defaultProvider := "test"
+	opts := auth.OIDCOptions{Providers: providers, DefaultProvider: &defaultProvider}
+	restTesterConfig := RestTesterConfig{
+		DatabaseConfig: &DbConfig{
+			OIDCConfig: &opts,
+			Unsupported: db.UnsupportedOptions{
+				OidcTestProvider: db.OidcTestProviderOptions{
+					Enabled: true,
+				},
+				OidcTlsSkipVerify: true,
+			},
+		}}
+	restTester := NewRestTester(t, &restTesterConfig)
+	restTester.SetAdminParty(false)
+	defer restTester.Close()
+
+	mockSyncGateway := httptest.NewTLSServer(restTester.TestPublicHandler())
+	defer mockSyncGateway.Close()
+	mockSyncGatewayURL := mockSyncGateway.URL
+	provider := restTesterConfig.DatabaseConfig.OIDCConfig.Providers.GetDefaultProvider()
+	provider.Issuer = mockSyncGateway.URL + "/db/_oidc_testing"
+	provider.CallbackURL = base.StringPtr(mockSyncGateway.URL + "/db/_oidc_callback")
+
+	// Send OpenID Connect request
```

Comment:
It wasn't immediately clear to me that we need to also set insecureSkipVerify on the test's HTTP client because both the _oidc and _oidc_testing endpoints are being run with the same TLSServer.  Maybe add that detail in this comment?

Reply:
I see, added comment to indicate that we need to also set insecureSkipVerify on the test's HTTP client because both the _oidc and _oidc_testing endpoints are being run with the same TLSServer.